# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1542128801 18000
#      Tue Nov 13 12:06:41 2018 -0500
# Node ID e142769913f8ad32b7f2f0ad13efa090a2982632
# Parent  b2a96efc88eb6d777082504da1f1b10f5bd21c90
move function declaration to correct header file

* fcn-info.h (get_dispatch_type): Move decl here from symtab.h (the
functions are defined in fcn-info.cc.

diff --git a/libinterp/corefcn/fcn-info.h b/libinterp/corefcn/fcn-info.h
--- a/libinterp/corefcn/fcn-info.h
+++ b/libinterp/corefcn/fcn-info.h
@@ -328,13 +328,20 @@ namespace octave
 
     octave_value dump (void) const { return m_rep->dump (); }
 
   private:
 
     std::shared_ptr<fcn_info_rep> m_rep;
   };
 
-  octave_value
+  extern OCTINTERP_API std::string
+  get_dispatch_type (const octave_value_list& args);
+
+  extern OCTINTERP_API std::string
+  get_dispatch_type (const octave_value_list& args,
+                     builtin_type_t& builtin_type);
+
+  extern octave_value
   dump_function_map (const std::map<std::string, octave_value>& fcn_map);
 }
 
 #endif
diff --git a/libinterp/corefcn/symtab.h b/libinterp/corefcn/symtab.h
--- a/libinterp/corefcn/symtab.h
+++ b/libinterp/corefcn/symtab.h
@@ -681,18 +681,11 @@ namespace octave
 
     // This function is generated automatically by mk-builtins.pl.
     void install_builtins (void);
   };
 
   extern bool out_of_date_check (octave_value& function,
                                  const std::string& dispatch_type = "",
                                  bool check_relative = true);
-
-  extern OCTINTERP_API std::string
-  get_dispatch_type (const octave_value_list& args);
-
-  extern OCTINTERP_API std::string
-  get_dispatch_type (const octave_value_list& args,
-                     builtin_type_t& builtin_type);
 }
 
 #endif
